src/v/cluster/cloud_metadata/offsets_upload_router.h

```diff
@@ -82,6 +83,14 @@ class offsets_upload_router
     ss::future<> start() { co_return; }
     ss::future<> stop() { return shutdown(); }
 
+    ss::future<offsets_upload_reply> request_upload(
+      offsets_upload_request req,
+      model::timeout_clock::duration timeout) override {
+        auto ntp = req.offsets_ntp;
+        co_return co_await process_or_dispatch(
+          std::move(req), std::move(ntp), timeout);
```

Comment:
might directly return the future from process_or_dispatch

Reply:
Will do